src/app/services/mail/mail.service.ts

```diff
@@ -157,13 +164,24 @@ export class MailService {
       })
 
       try {
+        let sendMailFromSG = false
+        const rand = Math.random() * 100
+        sendMailFromSG = rand <= config.nodemailer_client_threshold_sg
```

Comment:
/nit: there's no need to initialize `sendMailFromSG` as a non-const with a default. The assignment is systematic and guaranteed to yield a boolean:
```suggestion
        const rand = Math.random() * 100
        const sendMailFromSG = rand <= config.nodemailer_client_threshold_sg
```

Reply:
Also, since we have APM traces, could you try adding a tag of the US/SG decision into the trace span? That might allow easier searching and graphing later. Maybe something like this?

```typescript
const span = tracer.scope().active()
if (span) span.setTag('ses.region', sendMailFromSG ? 'sg', 'us')
```

I'm not sure if that should be on the outer span, or in the `nodemailer/sendMail` span btw 🤔 , Try with the outer span first, it might make searches easier (am not sure).

Documentation [here](https://docs.datadoghq.com/tracing/guide/add_span_md_and_graph_it)